CRM/Contribute/PseudoConstant.php

```diff
@@ -379,36 +372,25 @@ public static function &pcpStatus($column = 'label') {
   }
 
   /**
-   * Get all financial accounts for a Financial type.
+   * Get financial account for a Financial type.
    *
-   * The static array  $financialTypeAccount is returned
    *
-   *
-   * @param int $financialTypeId
-   * @param int $relationTypeId
-   * @return array
-   *   array reference of all financial accounts for a Financial type
+   * @param int $entityId
+   * @param string $accountRelationType
+   * @param string $entityTable
+   * @return int
    */
-  public static function financialAccountType($financialTypeId, $relationTypeId = NULL) {
-    if (!CRM_Utils_Array::value($financialTypeId, self::$financialTypeAccount)) {
-      $condition = " entity_id = $financialTypeId ";
-      CRM_Core_PseudoConstant::populate(
-        self::$financialTypeAccount[$financialTypeId],
-        'CRM_Financial_DAO_EntityFinancialAccount',
-        $all = TRUE,
-        $retrieve = 'financial_account_id',
-        $filter = NULL,
-        $condition,
-        NULL,
-        'account_relationship'
-      );
+  public static function getRelationalFinancialAccount($entityId, $accountRelationType, $entityTable = 'civicrm_financial_type') {
```

Comment:
I can't see any instance where $entityTable is passed in - should we remove this var?

Also - we should probably cache the values to Civi::statics - I think there are examples in CRM_Core_Pseudoconsant